File: Data Structures & Algorithms/products-of-array-discluding-self/submission-1.java
class Solution {
    public int[] productExceptSelf(int[] nums) {
        int[] ans = new int[nums.length];
        int product;
        for(int i = 0; i < nums.length; i++){
            product = 1;
            for(int j = 0; j < nums.length; j++){
                if(j != i){
                    product = product * nums[j];
                }
            }
            ans[i] = product;
        }
        return ans;
    }
}  

File: Data Structures & Algorithms/permutations/submission-1.py
class Solution:
    def permute(self, nums: List[int]) -> List[List[int]]:
        res = []
        used = set()
        perm = []
        def backtrack(pos):
            nonlocal res, perm, used
            if pos == len(nums):
                res.append(perm[:])
                return
            for num in nums:
                if num in used:
                    continue
                used.add(num)
                perm.append(num)
                backtrack(pos + 1)
                used.remove(num)
                perm.remove(num)

        backtrack(0)
        return res